### PR TITLE
feat(metrics_extraction): Drop old spec version

### PR DESCRIPTION
**Pull request by armenzg**
Now that we have migrated all alerts from not including the environment we can drop the old version.

### CODE REVIEW

File: tests/sentry/snuba/metrics/test_extraction.py [hunk 1]
```diff
@@ -699,26 +685,18 @@ def test_cleanup_with_environment_injection(query) -> None:
     transformed_query = f"({query}) AND (event.type:transaction)"
     environment = "production"
 
-    # We test with both new and old env logic, in this case queries should be identical in both logics since we
-    # scrape away parentheses.
-    for updated_env_logic in (True, False):
```
Comment (armenzg) [inline tests/sentry/snuba/metrics/test_extraction.py:704]: No need for this for loop anymore. The block left indents.

File: tests/sentry/snuba/metrics/test_extraction.py [hunk 2]
```diff
@@ -699,26 +685,18 @@ def test_cleanup_with_environment_injection(query) -> None:
...
-        assert spec.query_hash == transformed_spec.query_hash
+    spec = OnDemandMetricSpec(
+        field,
+        query,
+        environment=environment,
+    )
+    transformed_spec = OnDemandMetricSpec(
+        field,
+        transformed_query,
+        environment=environment,
```
Comment (armenzg) [inline tests/sentry/snuba/metrics/test_extraction.py:696]: We do not need to pass `spec_version=spec_version,` anymore.

File: tests/sentry/snuba/metrics/test_extraction.py [hunk 3]
```diff
@@ -699,26 +685,18 @@ def test_cleanup_with_environment_injection(query) -> None:
...
-            transformed_query,
-            environment=environment,
-            spec_version=spec_version,
-        )
-
-        assert spec.query_hash == transformed_spec.query_hash
+    spec = OnDemandMetricSpec(
+        field,
+        query,
+        environment=environment,
```
Comment (armenzg) [inline tests/sentry/snuba/metrics/test_extraction.py:691]: We do not need to pass spec_version=spec_version, anymore.

**Review by iambriccardo — COMMENTED** [review]
Thanks for your work! I left one comment

**Pull request closed without merging**